packages/vite-plugin/src/main.test.ts

```diff
@@ -1,15 +1,290 @@
+  })
+})
+
+describe('configureServer', { timeout: 15_000 }, () => {
+  const originalEnv = { ...process.env }
+  beforeEach(() => {
+    process.env = { ...originalEnv }
+  })
+
+  test('Populates Netlify runtime environment (globals and env vars)', async () => {
+    const fixture = new Fixture()
+      .withFile(
+        'vite.config.js',
+        `import { defineConfig } from 'vite';
+           import netlify from '@netlify/vite-plugin';
+
+           export default defineConfig({
+             plugins: [
+               netlify({ middleware: false })
+             ]
+           });`,
+      )
+      .withFile(
+        'index.html',
+        `<!DOCTYPE html>
+           <html>
+             <head><title>Hello World</title></head>
+             <body><h1>Hello from the browser</h1></body>
+           </html>`,
+      )
+    const directory = await fixture.create()
+    await fixture
+      .withPackages({
+        vite: '6.0.0',
+        '@netlify/vite-plugin': pathToFileURL(path.resolve(directory, PLUGIN_PATH)).toString(),
+      })
+      .create()
+
+    const { server } = await startTestServer({
+      root: directory,
+    })
+
+    expect((globalThis as Record<string, unknown>).Netlify).toBeInstanceOf(Object)
+    expect(process.env).toHaveProperty('NETLIFY_LOCAL', 'true')
+    expect(process.env).toHaveProperty('CONTEXT', 'dev')
+
+    await server.close()
+    await fixture.destroy()
+  })
+
+  describe('Middleware enabled', () => {
+    test('Returns static files from project dir', async () => {
+      const fixture = new Fixture()
+        .withFile(
+          'vite.config.js',
+          `import { defineConfig } from 'vite';
+             import netlify from '@netlify/vite-plugin';
+
+             export default defineConfig({
+               plugins: [
+                 netlify({
+                   middleware: true
+                 })
+               ]
+             });`,
+        )
+        .withFile(
+          'index.html',
+          `<!doctype html>
+            <html>
+              <head><title>Hello World</title></head>
+              <body>Hello from the static index.html file</body>
+              <script type="module" src="/js/main.js"></script>
+            </html>`,
+        )
+        .withFile(
+          'contact/email.html',
+          `<!doctype html>
+             <html>
+               <head><title>Contact us via email</title></head>
+               <body>Hello from another static file</body>
+             </html>`,
+        )
+        .withFile('js/main.js', `console.log('Hello from the browser')`)
+      const directory = await fixture.create()
+      await fixture
+        .withPackages({
+          vite: '6.0.0',
+          '@netlify/vite-plugin': pathToFileURL(path.resolve(directory, PLUGIN_PATH)).toString(),
+        })
+        .create()
+
+      const { server, url } = await startTestServer({
+        root: directory,
+      })
+
+      const response = await fetch(url)
```

Comment:
Note: all these `fetch` calls are replaced with playwright `page.goto()` calls in a follow-up PR, so we can ensure the rendered pages actually fully work end-to-end